tree/ntuple/v7/src/RField.cxx

```diff
@@ -1751,6 +1757,126 @@ void ROOT::Experimental::RClassField::AcceptVisitor(Detail::RFieldVisitor &visit
 
 //------------------------------------------------------------------------------
 
+std::size_t ROOT::Experimental::RField<TObject>::GetOffsetOfMember(const char *name)
+{
+   if (auto dataMember = TObject::Class()->GetDataMember(name)) {
+      return dataMember->GetOffset();
+   }
+   throw RException(R__FAIL('\'' + std::string(name) + '\'' + " is an invalid data member"));
+}
+
+ROOT::Experimental::RField<TObject>::RField(std::string_view fieldName)
+   : ROOT::Experimental::RFieldBase(fieldName, "TObject", ENTupleStructure::kRecord, false /* isSimple */)
+{
+#ifndef NDEBUG
+   static const auto cl = TObject::Class();
+#endif
+   assert(cl->GetClassVersion() == 1);
+   assert(cl->ClassProperty() & kClassHasExplicitCtor);
+   assert(cl->ClassProperty() & kClassHasExplicitDtor);
+   fTraits |= kTraitTriviallyConstructible | kTraitTriviallyDestructible;
```

Comment:
Is that correct or do we need the opposite?

Reply:
That is true of course, TObject is _not_ trivially constructible/destructible. Actually, this bit can be simplified to just test the class version.